src/PaymentGateways/Actions/UpdateDonationStatus.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Give\PaymentGateways\Actions;
+
+use Exception;
+use Give\Donations\Models\Donation;
+use Give\Donations\Models\DonationNote;
+use Give\Donations\ValueObjects\DonationStatus;
+use Give\Framework\PaymentGateways\Log\PaymentGatewayLog;
+use Give\Subscriptions\ValueObjects\SubscriptionStatus;
+
+/**
+ * @unreleased
+ */
+class UpdateDonationStatus
```

Comment:
Let's actually move this file and all the event handler files into the Framework/PaymentGateways folder as these are more of framework level actions.  The top level PaymentGateways directory is more about the specific gateways and the one inside framework is for shared logic.

Reply:
Resolved in 22eb283dd0d7805e1d4b43dea2b2da149f9ac5b8

The funny thing is that I started to develop it inside the framework folder, and then I changed it to keep similar to the recurring addon where similar classes live here: https://github.com/impress-org/give-recurring/tree/feature/mollie-submodule-GIVE-297/src/PaymentGatewayModules/EventHandlers

So, this makes me wonder if we should move these classes from the Recurring add-on to the Framework folder as well. This way, we can provide in the framework all necessary event handlers to implement a gateway, which can be useful for 3rd party integrations.

Reply:
@glaubersilva I like that idea! let's pull in the subscription event handlers into the framework folder as you suggested.

Reply:
Great! I just moved all subscription event handlers and their proper tests to Core.